tests/dashboards/api_tests.py

```diff
@@ -146,6 +151,53 @@ def create_dashboard_with_report(self):
             db.session.delete(dashboard)
             db.session.commit()
 
+    def test_get_dashboard_charts(self):
+        """
+        Dashboard API: Test getting charts belonging to a dashboard
+        """
+        self.login(username="admin")
+        admin = self.get_user("admin")
+        slice = self.insert_chart("slice1", [admin.id], 1, params="{}")
+        dashboard = self.insert_dashboard(
+            "title", "charted", [admin.id], admin, slices=[slice]
+        )
+        uri = f"api/v1/dashboard/{dashboard.id}/charts"
+        response = self.get_assert_metric(uri, "get_charts")
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.data.decode("utf-8"))
+        self.assertEqual(data["result"], [slice.data])
+        db.session.delete(dashboard)
+        db.session.delete(slice)
+        db.session.commit()
```

Comment:
I think you should handle dashboard and slice lifecycle(creation and destruction )  in the teardown after each test

Reply:
I agree. I was modeling this after the other tests around it, but have since learned about pytest fixtures. Will change.